i2c: nrfx_twi_rtio: Establish proper namespacing in driver

ubieda opened the pull request:
To clearly distinguish between common code and RTIO-only code on nrfx_twi and nrfx_twi_rtio.

Following-up on suggestion brought up on https://github.com/zephyrproject-rtos/zephyr/pull/72539#pullrequestreview-2067966324

ubieda committed:
i2c: nrfx_twi_rtio: Establish proper namespacing in driver

To clearly distinguish between common code and RTIO-only code
on nrfx_twi and nrfx_twi_rtio.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

masz-nordic commented on drivers/i2c/i2c_nrfx_twi_rtio.c:
```diff
@@ -122,24 +122,24 @@ static void event_handler(nrfx_twi_evt_t const *p_event, void *p_context)
-static const struct i2c_driver_api i2c_nrfx_twi_driver_api = {
+static const struct i2c_driver_api i2c_nrfx_twi_rtio_driver_api = {
 	.configure   = i2c_nrfx_twi_configure,
 	.transfer    = i2c_nrfx_twi_transfer,
```
```suggestion
	.transfer    = i2c_nrfx_twi_rtio_transfer,
```

anangl commented on drivers/i2c/i2c_nrfx_twi_rtio.c:
```diff
@@ -32,9 +32,9 @@ BUILD_ASSERT(
 
-static bool i2c_nrfx_twi_msg_start(const struct device *dev, uint8_t flags,
+static bool i2c_nrfx_twi_rtio_msg_start(const struct device *dev, uint8_t flags,
 				   uint8_t *buf, size_t buf_len, uint16_t i2c_addr)
```
Just nitpicking: the indentation here is a bit a bit weird now (also in a few more places below).

ubieda replied:
Thanks, I've added another commit to address non-related indentation changes. LMK if I missed anything

ubieda committed:
drivers: i2c: nrfx_twi_rtio: Minor indentation fixes

Not necessarily related to the namespacing changes, but just
establishing proper indentation formatting throughout the whole file.

Signed-off-by: Luis Ubieda <luisf@croxel.com>
